c/sudt_utils.h

```diff
@@ -22,7 +22,14 @@ void _sudt_id_to_key(const uint32_t account_id, uint8_t key[32]) {
   memcpy(key, (uint8_t *)&account_id, 4);
 }
 
-int _account_exists(gw_context_t *ctx, uint32_t account_id, bool* exists) {
+void _sudt_build_allowance_key(const uint32_t owner_account_id,
+                               const uint32_t spender_account_id,
+                               uint8_t key[32]) {
+  memcpy(key, (uint8_t *)&owner_account_id, 4);
```

Comment:
In the current construction, the allowance_key collision with the balance key if spender_account_id is 0.

I suggest using a fixed prefix in the allowance_key.
```
memcpy(key, b"allo", 4);
```